### PR TITLE
[Rebase #6730 and fix tests] Fix/UI settings overrides

### PR DESCRIPTION
**Pull request by ananzh**
### Description

This PR is originally from https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6730. We simply update the tests.

### From #6730 Description

There are a couple different fixes here, which can be viewed separately by looking at the separate commits. But the root cause of https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6704 was primarily the fact that the implementation of `getOverridesOrDefault` assumed the wrong object shape for the `overrides` property, when it tried to access a `value` property instead of just the value itself. Note that the existing unit test also had this same error.

Because the `ui_render_mixin` is an old legacy `.js` file, there's not much help from typescript, so I also added some enforcement to make sure we're always sending boolean values to the template.

Additionally, I realized that the validation for the `theme:version` setting was not correct, and needed updating to match our current themes.

Finally, as requested, I updated the new user theme controls to be deactivated by default.

No changelog necessary, as the [previous PR this fixes](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5652) has not yet been released.

### Future work

@AMoo-Miki @manasvinibs - The [past discussion](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/4475/files#r1251126091) and this [TODO comment](https://github.com/AMoo-Miki/OpenSearch-Dashboards/blob/31e8481fd2c0bf185d34c1140724e9320f6a676c/src/core/server/ui_settings/ui_settings_config.ts#L50-L53) seem to indicate that we don't actually want to maintain support for these overrides long term. But even if we do, I don't think the schema definition for overrides makes much sense:

```
overrides: schema.object({}, { unknowns: 'allow' }),
```
We probably want users to be aware of if they specify invalid overrides and not silently ignore (which is what was happening with an incorrectly set value in @BionIT 's configuration.

### Issues Resolved
Fixes https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6704

## Changelog
- skip

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by ruanyl — DISMISSED** [review]
Code changes look good to me, I'm just curious where to find `theme:enableUserControl` on the UI? Somehow I couldn't find it in advance settings page.

**Review by SuZhou-Joe — APPROVED** [review]

**Comment by ananzh** [comment]
> Code changes look good to me, I'm just curious where to find `theme:enableUserControl` on the UI? Somehow I couldn't find it in advance settings page.

Not sure. If you search `theme`, will you see it?
Screenshot 2024-08-21 at 9 08 23 AM

**Comment by ananzh** [comment]
@ruanyl @SuZhou-Joe I fixed the last test to make it more robust. Can you guys help me to re-approve it?

**Review by ruanyl — APPROVED** [review]

**Comment by andy-townsend** [comment]
> Code changes look good to me, I'm just curious where to find `theme:enableUserControl` on the UI? Somehow I couldn't find it in advance settings page.

Did you manage to resolve this @ruanyl ? I'm running 2.16 but can't see this under advanced settings either.

**Comment by joshuali925** [comment]
@andy-townsend the `enableUserControl` comes from a new theme feature introduced in #5652, it was reverted in 2.x by https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6978 due to some bugs and lack of test coverage, so it didn't get released in 2.16.

To release this feature, we should
1. add some integration tests
2. merge theme relevant PRs to main first: #7787, #7791
3. in 2.x undo #6978, then backport this PR and the two PRs mentioned above

**Comment by AMoo-Miki** [comment]
Decided against backporting.

File: src/core/server/ui_settings/ui_settings_config.ts [hunk 1]
```diff
@@ -58,7 +58,11 @@ const configSchema = schema.object({
   overrides: schema.object({}, { unknowns: 'allow' }),
   defaults: schema.object({
     'theme:darkMode': schema.maybe(schema.boolean({ defaultValue: false })),
-    'theme:version': schema.maybe(schema.string({ defaultValue: DEFAULT_THEME_VERSION })),
+    'theme:version': schema.maybe(
+      schema.oneOf([schema.literal('v7'), schema.literal('Next (preview)')], {
```
Comment (zhongnansu) [inline src/core/server/ui_settings/ui_settings_config.ts:62]: what if user wants to use v8 and v9?

**Pull request merged**